CRM/Core/Page.php

```diff
@@ -199,6 +199,8 @@ public function run() {
     $pageTemplateFile = $this->getHookedTemplateFileName();
     self::$_template->assign('tplFile', $pageTemplateFile);
 
+    self::$_template->addExpectedTabHeaderKeys();
```

Comment:
I wonder if just to keep things together if this should be up at line 162? It seems to get rid of the notices either way. Was there a reason to put it here? I don't have a strong opinion either way since I'm not aware of any situation where it would make a difference.

Reply:
@demeritcowboy This is intentional. Line 162 (i.e. where `ensureVariablesAreAssigned` is called) is within the constructor. That is fine for `ensureVariablesAreAssigned`, which is just setting default values to be overridden later.

That's not what we're doing in `addExpectedTabHeaderKeys` - here we're modifying assigned variables just before rendering, and so the page's `run()` function has to have run. Therefore running `addExpectedTabHeaderKeys` in the constructor would be too early.

There is probably an argument to moving `ensureVariablesAreAssigned` to line ~200, just before the new call to `addExpectedTabHeaderKeys`, but I have not looked closely to see if there would be side-effects to that change.